### PR TITLE
NCL-5210 Handle buildSrc project. Refactor/rename to improve naming. …

### PR DESCRIPTION
**Pull request by rnc**
…Move dumpConfig to Configuration

### CODE REVIEW

File: analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java [hunk 1]
```diff
@@ -90,6 +91,11 @@ public void verifyGroovyInjection() throws IOException, URISyntaxException, Mani
                 .hasOnlyOneElementSatisfying(l -> assertThat(l).contains("2.0.15.Final-redhat-00001"));
         assertTrue(lines.stream().anyMatch(s -> s.contains("CustomVersion( '1.0.1.redhat-00002', project )")));
         assertTrue(systemOutRule.getLog().contains("Attempting to read URL"));
+
+        assertFalse(FileUtils.readFileToString(new File(projectRoot, "settings.gradle"), Charset.defaultCharset())
+                .contains("x-pack"));
+        assertTrue(FileUtils.readFileToString(new File(projectRoot, "settings.gradle"), Charset.defaultCharset())
```
Comment (geoand) [inline analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java]: I think these 2 statements can be made a little more compact

File: common/src/main/java/org/jboss/gm/common/Configuration.java [hunk 2]
```diff
@@ -133,4 +134,25 @@ public DependencyPrecedence convert(Method method, String input) {
             return DependencyPrecedence.valueOf(input.toUpperCase());
         }
     }
+
+    /**
+     * Dumps the current configuration to the supplied logger.
+     *
+     * @param logger the logger to dump the configuration to.
+     */
+    default void dumpCurrentConfig(Logger logger) {
```
Comment (geoand) [inline common/src/main/java/org/jboss/gm/common/Configuration.java]: :+1:

**Comment by geoand** [comment]
Aside from the minor comment I added, looks good to me. I would like @TomasHofman to take a look as well

File: analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java (hunk 1 above)
Comment (rnc) [reply analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java]: Done :-)

Comment (geoand) [reply analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/SimpleProjectWithCustomGroovyScriptFunctionalTest.java]: Love it!

**Comment by richm** [comment]
@jcantrill @lukas-vlcek fyi

**Pull request merged**